### PR TITLE
Fixes #2943 XAML has no Intellisense

### LINKED ISSUE #2943: XAML has no Intellisense
Analysis of XAML files broke. Need to fix it

### PR DESCRIPTION
**Pull request by zooba**
Fixes #2943 XAML has no Intellisense
Fixes Resolved property
Reduce impact of .NET build targets with properties instead of excluding the entire target
Adds mscorlib reference automatically if missing
Adds upgrade test
Improves warning for unhandled types in analysis

### CODE REVIEW

File: Python/Templates/IronPython/ProjectTemplates/Python/WpfProject/WpfApp.pyproj [hunk 1]
```diff
@@ -29,6 +29,10 @@
     </Content>
   </ItemGroup>
   <ItemGroup>
+    <Reference Include="mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35">
```
Comment (huguesv) [inline Python/Templates/IronPython/ProjectTemplates/Python/WpfProject/WpfApp.pyproj]: b77a5c561934e089

File: Python/Product/Analysis/Interpreter/Ast/AstPythonModule.cs [hunk 2]
```diff
@@ -130,6 +135,8 @@ sealed class AstPythonModule : IPythonModule, IProjectEntry, ILocatedMember {
             var walker = new AstAnalysisWalker(interpreter, ast, this, filePath, DocumentUri, _members, true, true);
             ast.Walk(walker);
             walker.Complete();
+
+            ParseErrors = parseErrors.ToArray();
```
Comment (huguesv) [inline Python/Product/Analysis/Interpreter/Ast/AstPythonModule.cs]: `parseErrors` can be null, causing ANE on this line

Repro with test `UnassignedClassMembers` and `OsPathMembers`

File: Python/Product/Analysis/Parsing/Parser.cs [hunk 3]
```diff
@@ -4197,8 +4211,25 @@ struct Name {
 
         private bool PeekTokenForOrAsyncFor {
             get {
-                return PeekToken(Tokens.KeywordForToken) ||
-                    (AllowAsyncAwaitSyntax && PeekToken(Tokens.KeywordAsyncToken) && PeekToken2()?.Kind == TokenKind.KeywordFor);
+                if (PeekToken(Tokens.KeywordForToken)) {
+                    return true;
+                }
+                if (AllowAsyncAwaitSyntax || (_langVersion >= PythonLanguageVersion.V37 && _inGeneratorExpression)) {
```
Comment (MikhailArkhipov) [inline Python/Product/Analysis/Parsing/Parser.cs]: Should `_langVersion >= PythonLanguageVersion.V37` be simply part of the `AllowAsyncAwaitSyntax` property?

File: Python/Product/Analysis/Parsing/Parser.cs [hunk 4]
```diff
@@ -4197,8 +4211,25 @@ struct Name {
 
         private bool PeekTokenForOrAsyncFor {
```
Comment (MikhailArkhipov) [inline Python/Product/Analysis/Parsing/Parser.cs]: Alternative is change it to `PeekTokenForOrAsyncFor (bool inGeneratorExpression)`. Up to you.

Comment (zooba) [reply Python/Product/Analysis/Parsing/Parser.cs]: I started going that way and didn't like itp

File: Python/Product/Analysis/Parsing/Parser.cs (hunk 3 above)
Comment (zooba) [reply Python/Product/Analysis/Parsing/Parser.cs]: No, because it's still not allowed everywhere, but only in generator expressions.

**Pull request merged**